### PR TITLE
Add StringBuilder.Append(ROM<char>)

### PR DESCRIPTION
**Pull request by Wraith2**
Addresses https://github.com/dotnet/corefx/issues/32348

Ref, Tests and PerTests in corefx PR

Adds a `StringBuilder.Append(ReadOnlyMemory<char> value)` method to avoid the situation where a user assumes memory will work just as well as span but has it use the object parameter overload which causes boxing and a ToString() call on the memory instance which allocates.

### CODE REVIEW

File: src/System.Private.CoreLib/shared/System/Text/StringBuilder.cs [hunk 1]
```diff
@@ -1211,6 +1211,15 @@ public StringBuilder Append(ReadOnlySpan<char> value)
             return this;
         }
 
+        public StringBuilder Append(ReadOnlyMemory<char> value)
+        {
+            if (value.Length > 0)
+            {
+                return Append(value.Span);               
+            }
```
Comment (stephentoub) [inline src/System.Private.CoreLib/shared/System/Text/StringBuilder.cs]: There's no need to check .Length. The whole implementation can just be `=> Append(value.Span);`

File: src/System.Private.CoreLib/System.Private.CoreLib.sln [hunk 2]
```diff
@@ -21,24 +21,22 @@ Global
...
+		{3DA06C3A-2E7B-4CB7-80ED-9B12916013F9}.Debug|amd64.ActiveCfg = Debug|x86
 		{3DA06C3A-2E7B-4CB7-80ED-9B12916013F9}.Debug|arm.ActiveCfg = Debug|arm
 		{3DA06C3A-2E7B-4CB7-80ED-9B12916013F9}.Debug|arm.Build.0 = Debug|arm
 		{3DA06C3A-2E7B-4CB7-80ED-9B12916013F9}.Debug|arm64.ActiveCfg = Debug|arm64
 		{3DA06C3A-2E7B-4CB7-80ED-9B12916013F9}.Debug|arm64.Build.0 = Debug|arm64
 		{3DA06C3A-2E7B-4CB7-80ED-9B12916013F9}.Debug|x86.ActiveCfg = Debug|x86
 		{3DA06C3A-2E7B-4CB7-80ED-9B12916013F9}.Debug|x86.Build.0 = Debug|x86
-		{3DA06C3A-2E7B-4CB7-80ED-9B12916013F9}.Release|amd64.ActiveCfg = Release|amd64
-		{3DA06C3A-2E7B-4CB7-80ED-9B12916013F9}.Release|amd64.Build.0 = Release|amd64
+		{3DA06C3A-2E7B-4CB7-80ED-9B12916013F9}.Release|amd64.ActiveCfg = Release|x86
```
Comment (stephentoub) [inline src/System.Private.CoreLib/System.Private.CoreLib.sln]: Please revert this file.

File: src/System.Private.CoreLib/shared/System/Text/StringBuilder.cs (hunk 1 above)
Comment (Wraith2) [reply src/System.Private.CoreLib/shared/System/Text/StringBuilder.cs]: done.

File: src/System.Private.CoreLib/System.Private.CoreLib.sln (hunk 2 above)
Comment (Wraith2) [reply src/System.Private.CoreLib/System.Private.CoreLib.sln]: that was a slip, pushed back the version from master.

Comment (jkotas) [reply src/System.Private.CoreLib/System.Private.CoreLib.sln]: The changes in this file are still there...

Comment (Wraith2) [reply src/System.Private.CoreLib/System.Private.CoreLib.sln]: apparently you have do git things at the commandline if you want them to actually work. done.

**Comment by jkotas** [comment]
@dotnet-bot test Windows_NT x64 Checked CoreFX Tests  please

**Pull request merged**